src/detect-engine-analyzer.c

```diff
@@ -861,6 +863,17 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
                 jb_close(js);
                 break;
             }
+            case DETECT_TCPMSS: {
+                const DetectU16Data *cd = (const DetectU16Data *)smd->ctx;
+
+                jb_open_object(js, "tcp_mss");
+                const char *flag = TcpmssModeToString(cd->mode);
+                jb_set_string(js, "mode", flag);
+                jb_set_uint(js, "value1", cd->arg1);
+                jb_set_uint(js, "value2", cd->arg2);
```

Comment:
> I'm gathering that if the rule has a range, that's when we'll see something in `cd->arg2`. Could you make it so that if we have a range, instead of `value1` and `value2`, we have `min` and `max`, and if we only have the other cases, where we only have arg1, we use `value`, instead?

That means, I'll have to write a switch-case part, right?

Also, I noticed when the rule has range, arg2 is still logged.
I mentioned it in the PR details.

Reply:
When the rule has a range, it makes sense that `arg2` is logged, as the range is between a min and a max. Here, I'm just suggesting we change the names. When it's for equal, from what I saw in your test, `value2` will be 0, so I think that in case we can not log that, and just focus on what is indeed present in the rule.

You can use a switch, or `if else` statements, whatever makes more sense for the data you'll have to handle... I'm thinking that here, since we likely only have `arg1` and `arg2` with the `range` mode, and `if` could suffice. But, as you've already been doing, tinker and double-check, as I can make mistakes and misinterpret things, too ;)

Reply:
> When the rule has a range, it makes sense that `arg2` is logged, as the range is between a min and a max. Here, I'm just suggesting we change the names. When it's for equal, from what I saw in your test, `value2` will be 0, so I think that in case we can not log that, and just focus on what is indeed present in the rule.
> 
> You can use a switch, or `if else` statements, whatever makes more sense for the data you'll have to handle... I'm thinking that here, since we likely only have `arg1` and `arg2` with the `range` mode, and `if` could suffice. But, as you've already been doing, tinker and double-check, as I can make mistakes and misinterpret things, too ;)

Sorry! I meant, when rule isn't range, arg2 is still logged.

I traced the logic down to `rust/src/detect/uint.rs`, line 67.

Reply:
> > Sorry! I meant, when rule isn't range, arg2 is still logged.
> > 
> It isn't logged, it's to complete the struct otherwise it's unacceptable in rust. It does not seem to serve any purpose in this case afaict.
> 
> > I traced the logic down to `rust/src/detect/uint.rs`, line 67.
> 
> 

Whenever I test for a rule that isn't in range, `arg2` which don't have any value is logged in matches.
So I thought arg2 could be given the NULL value so it's not logged.

Reply:
`arg2` is template type in its definition which evaluates to u16 in this case. u16 cannot be NULL :wink:

> arg2 which don't have any value is logged in matches

only because you add it. You can check for the operand type and not log arg2 where it does not make sense (which seems everywhere but the range).

Reply:
Means I'd have to write a switch-case for it.

OK.
Thank you 😊.

Reply:
Why not an if statement as its an exclusive case? Do more operands use arg2?